File: 0013_Roman to Integer/RomantoInteger.cs
public class Solution {
    public int RomanToInt(string s) {
        if(String.IsNullOrEmpty(s)) return 0;
        
        var dict = new Dictionary<char,int>();
        dict.Add('I',1);
        dict.Add('V',5);
        dict.Add('X',10);
        dict.Add('L',50);
        dict.Add('C',100);
        dict.Add('D',500);
        dict.Add('M',1000);
        
        var ans = 0;
        var index = 0;
        while(index < s.Length){
            if(index + 1 < s.Length){
                
                int currVal = dict[s[index]];
                int nextVal = dict[s[index+1]];
                
                if(currVal >= nextVal){
                    ans += currVal;
                }else{
                    ans += (nextVal - currVal);
                    index++;
                }
                
                index++;
                
            }else{
                ans += dict[s[index++]];
            }
        }
        
        return ans;
        
    }
}

File: 0013_Roman to Integer/RomantoInteger.py
class Solution:
  def romanToInt(self, s):
    dict = {'I' : 1, 'V': 5, 'X': 10, 'L' : 50, 'C' : 100, 'D': 500, 'M': 1000}
    ans = dict[s[0]]
    for c, p in zip(s[1:], s[0:-1]):
      ans += dict[c]
      if dict[c] > dict[p]: ans -= 2 * dict[p]
    return ans